File: test/four/app/src/main/java/com/z3d/material/TexItem.java
package com.z3d.material;

public class TexItem {
    private int _id;
    public int get_id() {
        return _id;
    }
    public void set_id(int _id) {
        this._id = _id;
        this.name = "fs"+_id;
    }
    public String url;
    public TextureRes textureRes;
    public boolean isDynamic;
    public String  paramName;
    public boolean  isParticleColor;
    public boolean isMain;
    public  int type;
    public String  name;
    public int  wrap;
    public  int filter;
    public  int mipmap;

    public static int LIGHTMAP = 1;
    public static int LTUMAP = 2;
    public static int CUBEMAP = 3;
    public static int HEIGHTMAP = 4;
    public static int REFRACTIONMAP = 5;



}

File: metalios/iosgl/iosgl/pan3d/material/TexItem.m
//
//  TexItem.m
//  iosgl
//
//  Created by zhao on 9/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "TexItem.h"
@interface TexItem()
@property (nonatomic, assign)  int  idx;
@end
@implementation TexItem

+(int)LIGHTMAP;
{
    return 1;
}
+(int)LTUMAP;
{
    return 2;
}
+(int)CUBEMAP;
{
    return 3;
}
+(int)HEIGHTMAP;
{
    return 4;
}
+(int)REFRACTIONMAP;
{
    return 5;
}
 

- (instancetype)init
{
    self = [super init];
    if (self) {
         
    }
    return self;
}
-(void)setId:(int)id;
{
    _idx=id;
    self.name =[NSString stringWithFormat:@"fs%d",_idx];
}
-(int)id;
{
    return _idx;
}
@end
